MultiSource/Applications/ClamAV/CMakeLists.txt

```diff
@@ -178,9 +178,11 @@ if(TARGET_OS STREQUAL "Linux")
   find_package(Intl REQUIRED)
   target_link_libraries(clamscan ${Intl_LIBRARIES})
 endif()
-# zlib fails to build with correctly defined target OS macros.
-# (https://github.com/madler/zlib/pull/895)
-# Disable the compiler extension to workaround the build failure until a zlib
-# source update with the fix.
-target_compile_options(clamscan PRIVATE -fno-define-target-os-macros)
+if(APPLE)
+  # zlib fails to build with correctly defined target OS macros.
+  # (https://github.com/madler/zlib/pull/895)
+  # Disable the compiler extension to workaround the build failure until a zlib
+  # source update with the fix.
+  target_compile_options(clamscan PRIVATE -fno-define-target-os-macros)
```

Comment:
It would be better to use `CheckCXXCompilerFlag`, since there are versions of clang out there that don't support this flag.

https://cmake.org/cmake/help/latest/module/CheckCXXCompilerFlag.html

Reply:
Thank you for your suggestion! This is indeed a better solution.